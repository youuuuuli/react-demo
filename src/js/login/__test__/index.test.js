import React from 'react';
import { render } from '@testing-library/react';
import Login from '../index';

const useHistory = require('../../utils/useHistory');

let path = '';

describe('登入頁', () => {
  beforeEach(() => {
    useHistory.default = jest.fn().mockReturnValue({
      forward: jest.fn((url) => {
        path = url;

        return path;
      }),
    });
  });

  it('基本', () => {
    const { container } = render(<Login />);

    expect(container).toMatchSnapshot();
  });
});
