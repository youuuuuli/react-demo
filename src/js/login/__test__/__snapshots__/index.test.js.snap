// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`登入頁 基本 1`] = `
<div>
  <div
    class="ui centered grid login-page"
  >
    <div
      class="column"
    >
      <div
        class="ui right aligned container hide"
      >
        <img
          class="ui inline image lang-icon-cr"
          src="/lan_tw.png"
          title="繁體中文"
        />
        <img
          class="ui inline image lang-icon-cr"
          src="/lan_cn.png"
          title="简体中文"
        />
      </div>
      <h1
        class="ui center aligned header"
      >
        <div
          class="content"
        />
        <div
          class="sub header"
        />
      </h1>
      <form
        class="ui large form"
      >
        <div
          class="ui stacked segment"
        >
          <div
            class="field"
          >
            <div
              class="ui fluid left icon input"
            >
              <input
                type="text"
                value=""
              />
              <i
                aria-hidden="true"
                class="user icon"
              />
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui fluid left icon input"
            >
              <input
                type="password"
                value=""
              />
              <i
                aria-hidden="true"
                class="lock icon"
              />
            </div>
          </div>
          <button
            class="ui large fluid button submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
